### ISSUE TITLE
Ambilight+Hue multiple yeelight bug

### ISSUE THREAD

**Issue by GaoMax** [diyhue/diyHue#489]
**Describe the bug**
When setting up Amblight + Hue with multiple Yeelight devices, only one of the Yeelights work in entertainment mode at the time, the others stay in there default color. (tested with 2 strips + 2 bulbs)

**Expected behavior**
All bulbs should sync, I believe only on of the yeelight devices gets put into music mode, the one in music mode works, the ones that didn't get put into music mode don't work. The selection of which bulb gets put into music mode seems random.

2020-11-05 13:17:38,109 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.173 is now in music mode

**Logs**
`2020-11-05 13:17:35,501 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3 HTTP/1.1" 200 -
2020-11-05 13:17:36,320 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,328 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,378 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,428 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,478 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,494 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.52
2020-11-05 13:17:36,528 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,583 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,632 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,685 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,732 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:36,802 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/groups/4 HTTP/1.1" 200 -
2020-11-05 13:17:36,965 - WebServer - DEBUG - 192.168.188.30 - "PUT /api/93cb17da1f6611ebb6e7dca6323f8fb3/groups/4 HTTP/1.1" 200 -
2020-11-05 13:17:36,966 - webServer.server - INFO - in PUT method
2020-11-05 13:17:36,966 - webServer.server - INFO - /api/93cb17da1f6611ebb6e7dca6323f8fb3/groups/4
2020-11-05 13:17:36,967 - webServer.server - INFO - b'{"stream":{"active":false}}'
2020-11-05 13:17:36,967 - webServer.server - INFO - stop hue entertainent
entertain-srv: no process found
2020-11-05 13:17:37,281 - webServer.server - INFO - [
    {
        "success": {
            "/groups/4/stream": {
                "active": false
            }
        }
    }
]
2020-11-05 13:17:37,291 - WebServer - DEBUG - 192.168.188.30 - "PUT /api/93cb17da1f6611ebb6e7dca6323f8fb3/groups/4 HTTP/1.1" 200 -
2020-11-05 13:17:37,291 - webServer.server - INFO - in PUT method
2020-11-05 13:17:37,292 - webServer.server - INFO - /api/93cb17da1f6611ebb6e7dca6323f8fb3/groups/4
2020-11-05 13:17:37,292 - webServer.server - INFO - b'{"stream":{"active":true}}'
2020-11-05 13:17:37,293 - webServer.server - INFO - start hue entertainment

  . Seeding the random number generator... ok
  . Loading the CA root certificate ... ok (0 skipped)
  . Loading the server cert. and key... ok
  . Bind on udp://*:2100/ ... ok
  . Setting up the SSL/TLS structure... ok
  . Waiting for a remote connection ...2020-11-05 13:17:37,597 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.52
 ok
  . Performing the SSL/TLS handshake... hello verification requested
  . Waiting for a remote connection ... ok
  . Performing the SSL/TLS handshake... ok
    [ Protocol is DTLSv1.2 ]
    [ Ciphersuite is TLS-PSK-WITH-AES-128-GCM-SHA256 ]
    [ Record expansion is 37 ]
    [ Maximum fragment length is 16384 ]
  . Verifying peer X.509 certificate... ok
2020-11-05 13:17:37,806 - webServer.server - INFO - [
    {
        "success": {
            "/groups/4/stream": {
                "active": true
            }
        }
    }
]
2020-11-05 13:17:38,109 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.173 is now in music mode
2020-11-05 13:17:38,600 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.52
2020-11-05 13:17:39,502 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.97
2020-11-05 13:17:39,541 - WebServer - DEBUG - 192.168.188.97 - "GET /description.xml HTTP/1.1" 200 -
2020-11-05 13:17:40,798 - WebServer - DEBUG - 192.168.188.30 - "GET /api/93cb17da1f6611ebb6e7dca6323f8fb3/lights/1 HTTP/1.1" 200 -
2020-11-05 13:17:41,705 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.35
2020-11-05 13:17:42,175 - lightManager.core.lightRequest - INFO - sync with lights
`

**Additional context**
Tested with master and master-refactor. Problem occurs on both branches.

**Comment by alexyao2015** [comment]
So this is an issue with entertainment mode on yeelights? Please test with normal entertainment mode

**Comment by GaoMax** [comment]
Using entertainment mode in the Hue Essentials App works.

`2020-11-05 17:57:42,944 - WebServer - DEBUG - 192.168.188.53 - "PUT /api/hueess261f6311eb838edca6323f8fb3/groups/5 HTTP/1.1" 200 -
2020-11-05 17:57:42,945 - webServer.server - INFO - in PUT method
2020-11-05 17:57:42,945 - webServer.server - INFO - /api/hueess261f6311eb838edca6323f8fb3/groups/5
2020-11-05 17:57:42,946 - webServer.server - INFO - b'{"stream":{"active":true}}'
2020-11-05 17:57:42,946 - webServer.server - INFO - start hue entertainment

  . Seeding the random number generator... ok
  . Loading the CA root certificate ... ok (0 skipped)
  . Loading the server cert. and key... ok
  . Bind on udp://*:2100/ ... ok
  . Setting up the SSL/TLS structure... ok
  . Waiting for a remote connection ...2020-11-05 17:57:43,458 - webServer.server - INFO - [
    {
        "success": {
            "/groups/5/stream": {
                "active": true
            }
        }
    }
]
 ok
  . Performing the SSL/TLS handshake... hello verification requested
  . Waiting for a remote connection ... ok
  . Performing the SSL/TLS handshake... ok
    [ Protocol is DTLSv1.2 ]
    [ Ciphersuite is TLS-PSK-WITH-AES-128-GCM-SHA256 ]
    [ Record expansion is 37 ]
    [ Maximum fragment length is 16384 ]
  . Verifying peer X.509 certificate... ok
2020-11-05 17:57:43,872 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.134 is now in music mode
2020-11-05 17:57:44,142 - WebServer - DEBUG - 192.168.188.53 - "GET /api/hueess261f6311eb838edca6323f8fb3 HTTP/1.1" 200 -
2020-11-05 17:57:45,227 - lightManager.core.lightRequest - INFO - sync with lights
2020-11-05 17:57:45,902 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.82 is now in music mode
2020-11-05 17:57:47,213 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.173 is now in music mode
2020-11-05 17:57:47,228 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.92 is now in music mode
2020-11-05 17:57:47,896 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.97
2020-11-05 17:57:48,687 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:57:48,898 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.35
2020-11-05 17:57:49,351 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:57:49,773 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.134 is now in music mode
2020-11-05 17:57:49,900 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.35
2020-11-05 17:57:49,933 - WebServer - WARNING - 192.168.188.35 - code 404, message not found
2020-11-05 17:57:49,934 - WebServer - WARNING - 192.168.188.35 - "GET /description.xmlscription.xml HTTP/1.1" 404 -
2020-11-05 17:57:50,166 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
{"ok"}
2020-11-05 17:57:50,724 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.82 is now in music mode
2020-11-05 17:57:51,086 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:57:51,636 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.173 is now in music mode
2020-11-05 17:57:52,583 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.92 is now in music mode
2020-11-05 17:57:54,038 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:57:54,587 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.134 is now in music mode
2020-11-05 17:57:56,044 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:57:56,831 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:57:57,378 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.92 is now in music mode
2020-11-05 17:57:59,255 - protocols.yeelight - INFO - Yeelight device with IP 192.168.188.82 is now in music mode
2020-11-05 17:57:59,289 - functions.ssdp - DEBUG - Sending M-Search response to 192.168.188.97
2020-11-05 17:57:59,895 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
2020-11-05 17:58:00,687 - protocols.yeelight - WARNING - Yeelight command error: [Errno 32] Broken pipe
 connection was closed gracefully
  . Closing the connection... done
`

**Comment by alexyao2015** [comment]
So there are also problems with normal entertainment mode?

**Comment by GaoMax** [comment]
No, no problems with normal Entertainment mode. The logs indicate that with normal entertainment mode protocols.yeelight - INFO - Yeelight device with IP is now in music mode gets started for every light separately and with Ambilight music mode only gets started for one device.

**Comment by GaoMax** [comment]
Could it be that the Bug is on Ambilights side? or maybe a protocol difference from starting with entertainment mode? Is it possible to see what bulbs are a part of "/groups/2/stream"? If the bulbs are not present in this group, then it should be clear that the bug is not that the yeelights don't get put into music mode, but rather ambiligt not putting the together correctly.

**Comment by itsbasix** [comment]
Happened for me as well. It actually happens also with only genuine Hue Bulbs, so I do not think it is Yeelight related, but more Ambilight+Hue related. 
Only one light works, generally the first one in the list that is listed (alfabetical order?) by the Ambilight+Hue setup on the tv.
I tried with:
- 2 hue bulbs and 1 Yeelight, only Yeelight worked. 
- 2 hue bulbs : only 1 hue bulb works 

Also I noticed that from the Hue app, controlling the Yeelight from the main window requires double press for an action to happen (i.e light is off, you must press on, then off, then on again to make it light up)
